apm-agent-core/src/test/java/co/elastic/apm/agent/collections/GlobalThreadLocalTest.java

```diff
@@ -34,16 +36,20 @@
 
     @Test
     void setNullValueShouldNotThrow() {
-        GlobalThreadLocal.get(GlobalThreadLocalTest.class, "setNullValueShouldNotThrow").set(null);
+        WeakMaps
+            .buildThreadLocal()
+            .asGlobalThreadLocal(GlobalThreadLocalTest.class, "setNullValueShouldNotThrow")
+            .withDefaultValueSupplier(null)
+            .build();
```

Comment:
this tests only tries to use a `null` default value supplier, which is not exactly the same meaning as the previous test that tried to set a `null` value.